File: service/TFC.Application.DTO/Entity/ExerciseDTO.cs
using TFC.Domain.Model.Enum;

namespace TFC.Application.DTO.Entity
{
    public class ExerciseDTO
    {
        public long ExerciseId { get; set; }
        public string? ExerciseName { get; set; }
        public int? Sets { get; set; }
        public int? Reps { get; set; }
        public double? Weight { get; set; }
        public WeekDay DayName { get; set; }
    }
}

File: tfc_gym_app/lib/data/models/dto/exercise_dto.dart
import 'package:json_annotation/json_annotation.dart';
import '../enums/week_day.dart';

part 'exercise_dto.g.dart';

@JsonSerializable()
class ExerciseDTO {
  final int exerciseId;
  final String? exerciseName;
  final int? sets;
  final int? reps;
  final double? weight;
  final WeekDay? dayName;

  const ExerciseDTO({
    required this.exerciseId,
    this.exerciseName,
    this.sets,
    this.reps,
    this.weight,
    this.dayName,
  });

  factory ExerciseDTO.fromJson(Map<String, dynamic> json) => _$ExerciseDTOFromJson(json);
  Map<String, dynamic> toJson() => _$ExerciseDTOToJson(this);
}